config/initializers/app_artifacts.rb

```diff
@@ -1,10 +1,15 @@
 require 'app_artifacts'
 
 AppArtifacts.setup do |store|
+  # SAML keys and certs
   store.add_artifact(:saml_2019_cert, '/%<env>s/saml2019.crt')
   store.add_artifact(:saml_2019_key, '/%<env>s/saml2019.key.enc')
   store.add_artifact(:saml_2020_cert, '/%<env>s/saml2020.crt')
   store.add_artifact(:saml_2020_key, '/%<env>s/saml2020.key.enc')
   store.add_artifact(:saml_2021_cert, '/%<env>s/saml2021.crt')
   store.add_artifact(:saml_2021_key, '/%<env>s/saml2021.key.enc')
+
+  # OIDC keys
+  store.add_artifact(:oidc_private_key, '/%<env>s/oidc.key')
```

Comment:
maybe we could add an extra lil argument in our DSL here for a constructor? Just riffing:

```suggestion
  store.add_artifact(:oidc_private_key, '/%<env>s/oidc.key') do |value|
    OpenSSL::PKey::RSA.new(value)
  end
```

and so `add_artifact` we could modify like this, and boom, then we have typed values in our artifact struct now

```diff
diff --git a/lib/app_artifacts.rb b/lib/app_artifacts.rb
index 51bb2bd47..d0dee8907 100644
--- a/lib/app_artifacts.rb
+++ b/lib/app_artifacts.rb
@@ -25,8 +25,12 @@ class AppArtifacts
 
     # @param [Symbol] name
     # @param [String] path
-    def add_artifact(name, path)
+    # @yield [value] an optional block that can transform the raw value to a more common format,
+    #   for example parsing the value
+    # @yieldparam [String] value
+    def add_artifact(name, path, &block)
       value = read_artifact(path)
       raise MissingArtifactError.new("missing artifact: #{path}") if value.nil?
+      value = yield value if block_given?
       @artifacts[name] = value
       nil
```